### ISSUE TITLE
Best Buy Browser Disconnect after 10 mins

### ISSUE THREAD

**Issue by shark1987** [jef/nvidia-snatcher#405]
### Description
Just started getting this error from bestbuy after about 10 mins of running the script. First time was around 12:11PM CST, so possibly a new 10 minute block they put in place at noon.

The major problem is it does not recover and does not restart whatever "browser" disconnected. And it stops checking for all stores.

[12:11:19 PM] info :: ✖ [nvidia-api] [nvidia (3080)] founders edition :: OUT OF STOCK
[12:11:25 PM] info :: ✖ [bestbuy] [nvidia (3080)] founders edition :: OUT OF STOCK
[12:11:30 PM] info :: ✖ [nvidia-api] [nvidia (3080)] founders edition :: OUT OF STOCK
[12:11:39 PM] error :: ✖ [bestbuy] nvidia founders edition - Navigation failed because browser has disconnected!
[12:11:39 PM] error :: Navigation failed because browser has disconnected!
[12:11:39 PM] error :: Protocol error (Fetch.disable): Session closed. Most likely the page has been closed.

(Had to ctrl-c and restart script manually, I didn't notice the script wasn't running for almost an hour, so first timestamp from this run was [12:58:49 PM])

[1:09:32 PM] info :: ✖ [bestbuy] [nvidia (3080)] founders edition :: OUT OF STOCK
[1:09:35 PM] info :: ✖ [nvidia-api] [nvidia (3080)] founders edition :: OUT OF STOCK
[1:09:41 PM] info :: ✖ [nvidia-api] [nvidia (3080)] founders edition :: OUT OF STOCK
[1:09:46 PM] error :: ✖ [bestbuy] nvidia founders edition - Navigation failed because browser has disconnected!
[1:09:46 PM] error :: Protocol error (Fetch.disable): Session closed. Most likely the page has been closed.

### Possible solution
Add logic to restart this browser so the script can continue

**Comment by jef** [comment]
Should be implemented... I'll look into it later.

https://github.com/jef/nvidia-snatcher/blob/eee6b8183aa30840c761ad0d7ba69199b69dacf3/src/index.ts#L70-L78

**Comment by vp-cshow** [comment]
I'm unable to reproduce this; although I'm only checking for bestbuy cards...

I'm using `PAGE_SLEEP_MIN/MAX` at 2500 and 5000 respectively on the latest main build.

**Comment by shark1987** [comment]
So it is also no longer happening for me.

I had realized about an hour ago I had accidentally closed my "caffeinate" terminal on my macbook. So I think it is likely that it was actually just my laptop going to sleep after 10 mins. I'll go ahead and close this, sorry for the false alarm.

**Comment by shark1987** [comment]
I did confirm after more testing that the issue was caused by my macbook going to sleep. On resume, nvidia-snatcher just does nothing. The errors above are from when sleep starts, on resume you just get no output or anything until you CTRL-C.

I guess it would be "nice" if nvidia-snatcher prevented sleep, I did find a module to do that if the project is interested:
https://www.npmjs.com/package/stay-awake
